add environmental sustainability section (#247) (#255)
CaptainSifff/paper_teaching-learning-RSE: competencies.md

such as open science, path dependence, and vendor lock-in.

### Addressing environmental sustainability within planetary limits

> The last two decades saw the limits in the attainable miniaturisation of
> transistor technology approaching, and top chip clock frequency mostly
> settled [@Sutter2005].

Reviewer: ```suggestion
The last two decades saw transistor technology approach the limits of attainable miniaturisation, 
and top chip clock frequency begin to plateau [@Sutter2005].
```

> Nevertheless, popular perception remains fairly unaware of
> that, and a misleading belief in limitless growth of computing capabilities
> (storage, computing power, transfer speed) is still widespread.

Reviewer: ```suggestion
Nevertheless, a misleading belief in limitless growth of computing capabilities
(storage, computing power, transfer speed) is still widespread within popular perception.
```

A practical consequence of this is an ever-growing demand for resources to cover

> the expanded need of storage and processing, with no clear deceleration in

Reviewer: ```suggestion
the expanding need of storage and processing, with no clear deceleration in
```

sight (e.g. the IEA estimates a doubling in data centres energy consumption from
2024 to 2026 [@IEA2024]). At the same time, current science is well aware of
several planetary boundaries being trespassed due to human activities
[@Richardson2023]. Data processing, storage and transfer account for a non
negligible fraction [@IEA2024]. Demands to move resource consumption to a

> sustainable rate are well justified and supported by science [@Sills2019].
> RSEs have the opportunity to contribute to this effort by, for example,

Reviewer: ```suggestion
sustainable rate are well justified and supported by science [@Sills2019].

RSEs have the opportunity to contribute to this effort by, for example,
```

choosing computationally adequate approaches (e.g. recognising where a

> well-proved statistical method may do well instead of a power-hungry AI model,

Reviewer: ```suggestion
proven statistical method may suffice in place of a power-hungry AI model,
```

or configuring a test pipeline to minimise redundancy), and embracing data
frugality measures (e.g. recognising sufficient resolution when sampling data

> for processing or storage). If past computational solutions were frugal because
> of technological limits, future ones should tend to that by virtue of an
> awareness of what adequate solutions may be.

Reviewer: ```suggestion
for processing or storage). If past computational solutions were frugal because
of technological limits, in future they should tend to that by virtue of an
awareness of what may be adequate.
```

A practical consequence of this is an ever-growing demand for resources to cover
the expanded need of storage and processing, with no clear deceleration in
sight (e.g. the IEA estimates a doubling in data centres energy consumption from
2024 to 2026 [@IEA2024]). At the same time, current science is well aware of
several planetary boundaries being trespassed due to human activities
[@Richardson2023]. Data processing, storage and transfer account for a non
negligible fraction [@IEA2024]. Demands to move resource consumption to a
sustainable rate are well justified and supported by science [@Sills2019].
RSEs have the opportunity to contribute to this effort by, for example,
choosing computationally adequate approaches (e.g. recognising where a
well-proved statistical method may do well instead of a power-hungry AI model,
or configuring a test pipeline to minimise redundancy), and embracing data
frugality measures (e.g. recognising sufficient resolution when sampling data
for processing or storage). If past computational solutions were frugal because
of technological limits, future ones should tend to that by virtue of an
awareness of what adequate solutions may be.

> To be helpful in these regards, RSEs may need basic skills in extrapolating

Reviewer 2: I wonder if we should move the last sentence to the "RSE tasks and responsibilities" section. It combines various competencies and we can use it refer back to this section

Author: Great idea!

Reviewer 2: > Great idea!

ok, will provide something

\acp{IDE}, \gls{static-analysis} tools, \glspl{design-pattern} and documentation
(for oneself and others).

The RSE needs to be able to choose appropriate algorithms and techniques
(\gls{MOD} and \gls{NEW}). Apart from the technical feasibility, this choice is
also informed by the values outlined in Section @sec:values. For example, the
RSE needs to be able to estimate resource usage (processing, memory and storage

> consumption). Resource usage has not only a financial price tag but also impacts
> environmental concerns via associated energy consumption.

Reviewer: ```suggestion
consumption). Resource usage has not only a direct financial price tag but also
environmental costs via associated energy consumption.
```
